### PR TITLE
feat(replacement) Load balance INSERT query to storage nodes to avoid overloading the first node of the replica set

### PR DESCRIPTION
**Pull request by fpacifici**
https://github.com/getsentry/snuba/pull/2025 sends replacement INSERT queries to individual storage nodes to keep the data consistent after resharding. Before that INSERT queries were going to the query node and the distributed table.

That always preferred the first node of each replica set. As a consequence we are increasing the INSERT statement per second node too much when we have a spike of small replacements in the queue.

This adds a round robin load balancer to pick a different node for each INSERT to spread queries on the cluster. 
This load balancer could actually be made generic and added to the native clickhouse module.

### CODE REVIEW

File: snuba/replacer.py [hunk 1]
```diff
@@ -30,6 +30,90 @@
 NODES_REFRESH_PERIOD = 10
 
 
+class ShardedConnectionPool(ABC):
```
Comment (evanh) [inline snuba/replacer.py:33]: Why did you add this base class? There is only one possible subclass.

**Review by evanh — APPROVED** [review]
Looks good to me, just one question.

File: snuba/replacer.py (hunk 1 above)
Comment (fpacifici) [reply snuba/replacer.py:33]: I am thinking about a round robin load balancer that ignore the write nodes, but I did not implement it yet. That would make two implementations. Also having the base class will make it more reusable as we move the load balancer back into the clickhouse access package.

Comment (evanh) [reply snuba/replacer.py:33]: Got it. Thanks.
